configure.ac

```diff
@@ -1705,44 +1705,62 @@ else
   if test -n "$GUILE_CONFIG" ; then
     if test x"$GUILE" = xyes; then
       AC_MSG_CHECKING([for guile executable])
-      if $GUILE_CONFIG info guile > /dev/null 2>&1; then
-        GUILE="`$GUILE_CONFIG info guile`"
+      # Try extracting it via guile-config first. If it's defined there it's the most reliable result
+      GUILE="`$GUILE_CONFIG info guile 2>/dev/null`"
+      if test -n "$GUILE";  then
         AC_MSG_RESULT([$GUILE])
       else
-        AC_MSG_RESULT([not in guile-config, constructing path])
+        AC_MSG_RESULT([not found via guile-config - constructing path])
         AC_MSG_CHECKING([for guile bindir])
         guile_bindir="`$GUILE_CONFIG info bindir`"
         AC_MSG_RESULT([$guile_bindir])
-        GUILE=$guile_bindir/guile
-        if ! test -f "$GUILE" ; then
-          GUILE=
-          AC_PATH_PROG(GUILE, guile)
-        fi
+        GUILE="$guile_bindir/guile"
+      fi
+      if ! test -f "$GUILE" ; then
+        GUILE=
+        AC_PATH_PROG(GUILE, guile)
       fi
     fi
 
-    if test -f "$GUILE" ; then
+    if ! test -f "$GUILE" ; then
+      AC_MSG_WARN([no suitable guile executable found. Disabling Guile])
+    else
       AC_MSG_CHECKING([for guile version])
       guile_version=`$GUILE -c '(display (effective-version))'`
       AC_MSG_RESULT([$guile_version])
       AC_MSG_CHECKING([for guile version >= 1.8])
       guile_good_version=`$GUILE -c '(if (>= (string->number (effective-version)) 1.8) (display "yes") (display "no"))'`
       AC_MSG_RESULT([$guile_good_version])
       if test x"$guile_good_version" != xyes ; then
+        AC_MSG_WARN([at least guile version 1.8 is required. Disabling Guile])
         GUILE=
       fi
     fi
 
-    if test -z "$GUILE_CFLAGS" ; then
-      AC_MSG_CHECKING([for guile compile flags])
-      GUILE_CFLAGS="`$GUILE_CONFIG compile`" # Note that this can sometimes be empty
-      AC_MSG_RESULT([$GUILE_CFLAGS])
+    if test -f "$GUILE" ; then
+      # Test if guile-config and guile versions match. They should.
+      gc_version="`$GUILE_CONFIG --version 2>&1 | head -1`"
+      gc_version="${gc_version##* }"
+      g_version="`$GUILE --version | head -1`"
+      g_version="${g_version##* }"
```

Comment:
We don't currently rely on `${foo##bar}` anywhere in SWIG.

It is specified by POSIX, but `info autoconf` warns:

```
'${#VAR}'
'${VAR%WORD}'
'${VAR%%WORD}'
'${VAR#WORD}'
'${VAR##WORD}'
     Posix requires support for these usages, but they do not work with
     many traditional shells, e.g., Solaris 10 '/bin/sh'.
```

So probably prudent to use sed instead.  You can do the `head -1` at the same time, so it's no more external commands to run:

```
      gc_version="`$GUILE_CONFIG --version 2>&1 | sed 's/.* //;q'`"
      g_version="`$GUILE --version | sed 's/.* //;q'`"
```
```

Reply:
Good point. I have changed the relevant commit.